astroquery/eso/utils.py

```diff
@@ -0,0 +1,197 @@
+    if allowed_values:
+        where_constraints.append(_format_helper(allowed_values))
+
+    where_constraints += [
+        f"{k} {_adql_sanitize_op_val(v)}" for k, v in column_filters.items()
+    ]
+    return where_constraints
+
+
+def reorder_columns(table: Table,
+                    leading_cols: Optional[List[str]] = None):
+    """
+    Reorders the columns of the pased table so that the
+    colums given by the list leading_cols are first.
+    If no leading cols are passed, it defaults to
+    ['object', 'ra', 'dec', 'dp_id', 'date_obs']
+    Returns a table with the columns reordered.
+    """
+    if not isinstance(table, Table):
+        return table
+
+    leading_cols = leading_cols or DEFAULT_LEAD_COLS_RAW
+    first_cols = []
+    last_cols = table.colnames[:]
+    for x in leading_cols:
+        if x in last_cols:
+            last_cols.remove(x)
+            first_cols.append(x)
+    last_cols = first_cols + last_cols
+    table = table[last_cols]
+    return table
+
+
+def _adql_sanitize_op_val(op_val):
+    """
+    Expected input:
+        "= 5", "< 3.14", "like '%John Doe%'", "in ('item1', 'item2')"
+        or just string values like "ESO", "ALMA", "'ALMA'", "John Doe"
+
+    Logic:
+        returns "<operator> <value>" if operator is provided.
+        Defaults to "= <value>" otherwise.
+    """
+    supported_operators = ["<=", ">=", "!=", "=", ">", "<",
+                           "not like ", "not in ", "not between ",
+                           "like ", "between ", "in "]  # order matters
+
+    if not isinstance(op_val, str):
+        return f"= {op_val}"
+
+    op_val = op_val.strip()
+    for s in supported_operators:
+        if op_val.lower().startswith(s):
+            operator, value = s, op_val[len(s):].strip()
+            return f"{operator} {value}"
+
+    # Default case: no operator. Assign "="
+    value = op_val if (op_val.startswith("'") and op_val.endswith("'")) else f"'{op_val}'"
+    return f"= {value}"
+
+
+def raise_if_coords_not_valid(cone_ra: Optional[float] = None,
+                              cone_dec: Optional[float] = None,
+                              cone_radius: Optional[float] = None) -> bool:
+    """
+    ra, dec, radius must be either present all three
+    or absent all three. Moreover, they must be float
+    """
+    are_all_none = (cone_ra is None) and (cone_dec is None) and (cone_radius is None)
+    are_all_float = isinstance(cone_ra, (float, int)) and \
+        isinstance(cone_dec, (float, int)) and \
+        isinstance(cone_radius, (float, int))
+    is_a_valid_combination = are_all_none or are_all_float
+    if not is_a_valid_combination:
+        raise ValueError(
+            "Either all three (cone_ra, cone_dec, cone_radius) must be present or none.\n"
+            "Values provided:\n"
+            f"\tcone_ra = {cone_ra}, cone_dec = {cone_dec}, cone_radius = {cone_radius}"
+        )
+
+
+def _py2adql(user_params: _UserParams) -> str:
```

Comment:
can we call this something more descriptive? `parameters_to_adql` or something similar?

Reply:
Renamed to `_build_adql_string`